src/msw/joystick.cpp

```diff
@@ -37,8 +37,127 @@
 
 #include <regstr.h>
 
+// Use optimised count trailing zeros where available.
```

Comment:
This really doesn't seem like the best place for this, we ought to move it to some header (`wx/math.h` probably, as I don't see anything more appropriate) and rename it to `wxCountLeadingZeroes()` or something like this.

Reply:
Moving the updated definition to wx/math.h gives me a compiler warning:
```
In file included from ../src/common/extended.c:18:0:
../include/wx/math.h: In function 'wxCountTrailingZeroes':
../include/wx/math.h:187:5: warning: implicit declaration of function 'wxCHECK_MSG' [-Wimplicit-function-declaration]
     wxCHECK_MSG(x > 0, 0, "Undefined for x == 0.");
```
I guess I need to add an #include to wx/math.h, but which one?
 * include/wx/debug.h
 * interface/wx/debug.h

Reply:
This function should be inside C++-only section (the error you get is when compiling `extended.c`). But if you still have problems with this, just leave it here for now then and I'll move it to `wx/math.h` later.

Reply:
OK - I was attempting to use wxCHECK_MSG to prevent undefined usage of either the builtin or fallback, but looks like I misunderstood something. I'll leave it to you to find a more appropriate home for this code.